### PR TITLE
Omit empty and optional values in form data in Try It

### PR DESCRIPTION
**Pull request by GeliXav**
Draft PR for : https://github.com/stoplightio/elements/issues/1829

### CODE REVIEW

**Comment by Nezteb** [comment]
@GeliXav Hello! Just an FYI that we know your PR exists and we've not forgotten about it. We've been working on new features for Stoplight Studio and have been focused on that. We will make sure someone reviews your PR though!

**Comment by GeliXav** [comment]
Hey @Nezteb, any news on when this PR will be reviewed ? Thanks a lot !

**Comment by Nezteb** [comment]
@GeliXav Thank you for your patience on this and the ping.

I tested this out to get a demo video for my team:

https://user-images.githubusercontent.com/3588798/151081400-cfb36e2e-174e-4645-8752-6caa0f8be1e6.mov

Overall, I like the approach! I'm glad to see that you account for required fields. 😄

My only concern is the UI/UX of the checkboxes. I'm going to make sure our product manager and designer take a look and provide you with feedback. (@mnaumanali94 @noor-ahmad)

Thank you for the contribution! I will make sure we don't lose this one.

**Comment by Nezteb** [comment]
Okay @GeliXav, I got feedback!

In order to make it clear which fields the checkboxes are referring to, our designer suggested indenting them (i.e. adding some padding so that each checkbox is moved to the right a bit). Also changing the language from "Omit the field value" to just "Omit `<actual-field-name>`". These two things will make it extra clear for users!

Our designer also came up with a future design idea to clarify the UX even more:
temp

But you've already done a lot of good work and we'll happily merge the PR without the eye icons. 😄

**Comment by GeliXav** [comment]
Hey @Nezteb, I applied your suggestions outside of the eye icons :) !

File: packages/elements-core/src/components/TryIt/Body/request-body-utils.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {
 } from '../Parameters/parameter-utils';
 
 export type BodyParameterValues = Record<string, string | File>;
+export type ParamaterOptional = Record<string, boolean>;
```
Comment (jvyang) [inline packages/elements-core/src/components/TryIt/Body/request-body-utils.ts:13]: There's a typo here. Let's rename it.

```suggestion
export type ParameterOptional = Record<string, boolean>;
```

**Comment by Nezteb** [comment]
Screen Shot 2022-01-28 at 3 23 42 PM

Awesome @GeliXav, we love what you've done! I fixed that typo @jvyang mentioned too, so we're all good! I will merge!

**Review by Nezteb — APPROVED** [review]

**Comment by GeliXav** [comment]
Hey @Nezteb, thanks for spending time and effort on this. Any idea on when this will get merged ? I see a test not passing, but I don't think it's related to my changes.

**Comment by Nezteb** [comment]
@GeliXav Thank you for your patience! I reran the failed tests and they passed, but now there are merge conflicts. Another Stoplighter moved some CSS around in https://github.com/stoplightio/elements/commit/9f9ae13782edde922f4299a3db817a545095aa55, so I'll try to fix that for you!

**Comment by GeliXav** [comment]
@Nezteb Thanks a lot, I did the merge by myself, I think this time we should be good for a merge ! :D

**Pull request merged**